openedx_external_enrollments/external_enrollments/greenfig_external_enrollment.py

```diff
@@ -0,0 +1,119 @@
+"""GreenfigInstanceExternalEnrollment class file."""
+import io
+import logging
+
+import requests
+from rest_framework import status
+
+from openedx_external_enrollments.edxapp_wrapper.get_site_configuration import configuration_helpers
+from openedx_external_enrollments.edxapp_wrapper.get_student import get_user
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.models import EnrollmentRequestLog
+
+LOG = logging.getLogger(__name__)
+
+
+class GreenfigInstanceExternalEnrollment(BaseExternalEnrollment):
+    """
+    GreenfigInstanceExternalEnrollment class.
+    """
+
+    def __init__(self):
+        """
+        """
+        self.DROPBOX_API_URL = configuration_helpers.get_value('DROPBOX_API_URL', 'https://content.dropboxapi.com/2')
+        self.DROPBOX_FILE_PATH = configuration_helpers.get_value('DROPBOX_FILE_PATH', '/courses.txt')
+        self.DROPBOX_TOKEN = configuration_helpers.get_value('DROPBOX_TOKEN', 'token')
+
+    def __str__(self):
+        """
+        """
+        return "greenfig"
+
+    def _execute_post(self, url, headers=None, json_data=None):
+        """
+        """
+        response = requests.post(
+            url=url,
+            data=json_data,
+            headers=headers
+        )
+        return response
+
+    def _get_enrollment_headers(self):
+        """
+        """
+        headers = {
+            "Authorization": "Bearer {token}".format(token=self.DROPBOX_TOKEN),
+            "Content-Type": "application/octet-stream",
+            "Dropbox-API-Arg": "{\"path\":\"%s\",\"mode\":{\".tag\":\"overwrite\"}}" % self.DROPBOX_FILE_PATH
+        }
+
+        return headers
+
+    def _get_download_headers(self):
+        """
+        """
+        headers = {
+            "Authorization": "Bearer {token}".format(token=self.DROPBOX_TOKEN),
+            "Dropbox-API-Arg": "{\"path\":\"%s\"}" % self.DROPBOX_FILE_PATH
+        }
+
+        return headers
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        """
+        user, _ = get_user(email=data.get("user_email"))
+        dropbox_file = self._get_course_list(course_settings)
+        dropbox_file = dropbox_file.text.split('\n')
+        enrollment_data = u"{first_name}, {last_name}, {email}, {course_id}\n".format(
+            first_name=user.first_name,
+            last_name=user.last_name,
+            email=user.email,
+            course_id=course_settings.get("external_course_run_id"),
```

Comment:
Does this need to define a default value? In case there is no external_course_run_id? @luismorenolopera

Reply:
I think this should be rquired, it should not continue if external_course_run_id does not exist